src/components/auction/OrderbookContainer/index.tsx

```diff
@@ -0,0 +1,183 @@
+  display: flex;
+  align-items: ${(props) =>
+    props.alignItems
+      ? typeof props.alignItems !== 'string'
+        ? props.alignItems[0]
+        : props.alignItems
+      : 'flex-start'};
+  justify-content: space-between;
+  flex-direction: ${(props) =>
+    props.flexDir ? (typeof props.flexDir !== 'string' ? props.flexDir[0] : props.flexDir) : 'row'};
+  ${(props) =>
+    props.flexDir
+      ? props.flexDir === 'column' || props.flexDir[0] === 'column'
+        ? 'flex-wrap: nowrap'
+        : 'flex-wrap: wrap'
+      : 'flex-wrap: wrap'};
+  margin: ${(props) =>
+    props.margin ? (typeof props.margin !== 'string' ? props.margin[0] : props.margin) : '0'};
+  @media (min-width: 768px) {
+    align-items: ${(props) =>
+      props.alignItems
+        ? typeof props.alignItems !== 'string'
+          ? props.alignItems[1]
+          : props.alignItems
+        : 'flex-start'};
+    flex-direction: ${(props) =>
+      props.flexDir
+        ? typeof props.flexDir !== 'string'
+          ? props.flexDir[1]
+          : props.flexDir
+        : 'row'};
+    ${(props) =>
+      props.flexDir
+        ? props.flexDir === 'column' || props.flexDir[1] === 'column'
+          ? 'flex-wrap: nowrap'
+          : 'flex-wrap: wrap'
+        : 'flex-wrap: wrap'};
+    margin: ${(props) =>
+      props.margin ? (typeof props.margin !== 'string' ? props.margin[1] : props.margin) : '0'};
+  }
+`
+
+const SectionTitle = styled(PageTitle)`
+  margin-bottom: 0;
+  margin-top: 0;
+`
+
+const StyledButtonSelect = styled(ButtonSelect)`
+  height: 19px;
+  font-size: 13px;
+  border-color: ${({ theme }) => theme.textField.color};
+  padding: 2px 6px;
+  line-height: 13px;
+  cursor: pointer;
+  svg {
+    width: 8px;
+    margin-left: 7px;
+  }
+`
+
+const StyledDropdown = styled(Dropdown)`
+  margin-left: 16px;
+  @media (min-width: 768px) {
+    margin-left: 0;
+    margin-right: 16px;
+  }
+  &.isOpen {
+    button {
+      background-color: ${({ theme }) => theme.textField.color};
+      border-color: ${({ theme }) => theme.textField.color};
+      color: ${({ theme }) => theme.textField.backgroundColor};
+      .fill {
+        fill: ${({ theme }) => theme.textField.backgroundColor};
+      }
+    }
+  }
+  .dropdownItems {
+    min-width: 82px;
+  }
+`
+
+const StyledDropdownItem = styled(DropdownItem)`
+  padding: 5px 13px;
+  font-size: 13px;
+  font-weight: normal;
+  line-height: 1.08;
+  justify-content: space-between;
+`
+
+const StyledCheckbox = styled(Checkbox)`
+  border-radius: 50%;
+  background-color: ${({ theme }) => theme.textField.color};
+  &:before {
+    border-radius: 50%;
+  }
+`
+
+// TODO: Build the array dynamically based on the current bids
+const granularityOptions = ['0.001', '0.01', '0.1', '1', '10', '50', '100']
+const getClosestNumber = (numbers: string[], goal: number) => {
+  return numbers.reduce(function (prev, curr) {
+    return Math.abs(Number(curr) - goal) < Math.abs(Number(prev) - goal) ? curr : prev
+  })
+}
```

Comment:
Yes, that's correct! The only thing to fix would be

```javascript
const min = Math.min(diffs)
```
to this
```javascript
const min = Math.min(...diffs)
```
You approach is easier to follow but the only drawback is that the numbers array could potentially be looped 3 times instead of 1 using reduce

Reply:
Yep, good point, still linear but 3x worse.